app/services/set_decline_by_default.rb

```diff
@@ -0,0 +1,33 @@
+class SetDeclineByDefault
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call
+    return false unless decided_choices.count == 3 && decided_choices.any?(&:offered_at)
+
+    most_recent_decision_date = [
+      decided_choices.maximum(:offered_at),
+      decided_choices.maximum(:rejected_at),
+    ].compact.max
+
+    dbd_days = TimeLimitCalculator.new(
+      rule: :decline_by_default,
+      effective_date: most_recent_decision_date,
+    ).call
+
+    dbd_date = dbd_days.business_days.after(most_recent_decision_date).end_of_day
+
+    application_form.application_choices.all? do |application_choice|
+      application_choice.update(decline_by_default_at: dbd_date)
```

Comment:
It feels a bit odd to make the update as a side-effect of `all?`. How about `application_form.application_choices.update_all(...)`.

Reply:
yes, this smell a bit 👃